project/dataset/jeopardy_dataset.py

```diff
@@ -0,0 +1,105 @@
+import json
+from PIL import Image
+from torch.utils.data import Dataset
+import os
+import string
+from tqdm import tqdm
+
+class JeopardyDataset(Dataset):
+    """Face Landmarks dataset."""
+
+    def __init__(self, questions_file, answers_file, images_dir, transform):
+        """
+        Args:
+            questions_file (string): Path to the json file with questions.
+            answers_file (string): Path to the json file with annotations.
+            root_dir (string): Directory with all the images.
+            transform (callable, optional): Optional transform to be applied
+                on a sample.
+
+        Example call:
+            test_ds = JeopardyDataset("v2_OpenEnded_mscoco_train2014_questions.json", "v2_mscoco_train2014_annotations.json", "images")
+
+        """
+        q_f = open(questions_file, 'r')
+        a_f = open(answers_file, 'r')
+        
+        self.questions = json.load(q_f)["questions"]
+        self.answers = json.load(a_f)["annotations"]
+        self.images_dir = images_dir
+        self.image_id_to_filename = self._find_images()
+        self.transform = transform
+        self.question_to_answer = self._find_answers()
+        self.mega_dict = {} # this will store our indices
+        self.numerical = {}
+
+        answer_text = ' . '.join(list(self.question_to_answer.values()))
+        question_text = ' . '.join([q['question'] for q in self.questions])
+        question_text = question_text.replace("?", " END")
```

Comment:
So I was a little bit unsure about the choices surrounding this - I'd ideally just strip out all punctuation, but doesn't the question mark have special semantic meaning since we're dealing with questions half the time. Perhaps we can talk about this?